File: data/multilang_comparison/program_cpp_array/src/func3.cpp
#include "program_cpp_array.hpp" 
Array* func3(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array195;
   if (pCounter > 0) {
      array195 = vars->data[--pCounter];
      array195->refC++;
      DEBUG_COPY(array195->id);
   } else {
      array195 = new Array();
      array195->size = 57;
      array195->refC = 1;
      array195->id = 195;
      array195->data = new unsigned int[array195->size]();
      DEBUG_NEW(array195->id);
   }
   Array_param params0;
   params0.size = 1;
   Array* data_params0[1];
   params0.data = data_params0;
   params0.data[0] = array195;
   Array* array196 = func6(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array196->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   unsigned int loop56 = 0;
   unsigned int loopLimit56 = (50)/2 + 1;
   for(; loop56 < loopLimit56; loop56++) {
      for (int i = 0; i < array196->size; i++) {
         array196->data[i]++;
      }
   }
   for (int i = 0; i < array195->size; i++) {
       if (array195->data[i] == 91) {
           return array195;
       }
   }
   array195->refC--;
   if(array195->refC == 0) {
      delete[] array195->data;
      array195->data = nullptr;
      DEBUG_FREE(array195->id);
      delete array195;
   }
   return array196;
}

File: data/multilang_comparison/program_go/src/func3.go
package main
import "math/rand"
var _ = rand.Intn
func Func3(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array195 *Array
   if pCounter > 0 {
       array195 = vars.Data[pCounter-1]
       pCounter--
       array195.RefC++
   } else {
       array195 = &Array{
           Size: 57,
           RefC: 1,
           Id: 195,
           Data: make([]uint32, 57),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array195
   var array196 *Array = Func6(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   var loop56 int = 0;
   var loopLimit56 int = (50)/2 + 1;
   for ; loop56 < loopLimit56; loop56++ {
      for i := 0; i < array196.Size; i++ {
          array196.Data[i]++
      }
   }
   for i := 0; i < array195.Size; i++ {
       if array195.Data[i] == 91 {
           return array195
       }
   }
   array195.RefC--
   if array195.RefC == 0 {
       array195.Data = nil
       array195 = nil
   }
   return array196;
}

